draft-ietf-jsonpath-base.md

```diff
@@ -569,206 +581,226 @@ of node.
+
+#### Semantics
+{: numbered="false" toc="exclude"}
+
+The `dot-selector` selects the value corresponding to the member name from any JSON object. It selects no values from any other JSON value.
+
+Note that the `dot-selector` follows the philosophy of JSON strings and is
+allowed to contain bit sequences that cannot encode Unicode characters (a
+single unpaired UTF-16 surrogate, for example).
+The behaviour of an implementation is undefined for member names which do
+not encode Unicode characters.
+
+### Dot Wild Card Selector
 
 #### Syntax
+{: numbered="false" toc="exclude"}
 
-A union selector consists of one or more union elements.
+The dot wild card selector has the shape `'.*'`.
 
 ~~~~ abnf
-selector =/ union
-union = "[" ws union-elements ws "]" ; [...]
-ws = *" "                             ; zero or more spaces
-union-elements = union-element /
-                 union-element ws "," ws union-elements
-                                       ; ,-separated list
+dot-wild-selector    = "." "*"            ;  dot followed by asterisk
 ~~~~
 
-
 #### Semantics
+{: numbered="false" toc="exclude"}
 
-A union selects any node which is selected by at least one of the union selectors
-and selects the concatenation of the
-lists (in the order of the selectors) of nodes selected by the union elements.<!--  TODO: define whether duplicates are kept or removed.  -->
+A dot selector combined with a single asterisk is a wild card. It selects all member values of an object as well as all elements of an array. Applying the `dot-wild-selector` to primitive JSON values, as number, string, or literal, does nothing.
 
 
-#### Child {#unionchild}
+### Index Selector
 
-##### Syntax
+#### Syntax
 {: numbered="false" toc="exclude"}
 
-A child is a quoted string.
+An index selector `'[<index>]'` addresses a single object member value or a single array element value.
 
 ~~~~ abnf
-union-element = child ; see below for more alternatives
-child = %x22 *double-quoted %x22 / ; "string"
-        "'" *single-quoted "'"   ; 'string'
-
-double-quoted = dq-unescaped /
-          escape (
-              %x22 /         ; "    quotation mark  U+0022
-              "/" /          ; /    solidus         U+002F
-              "\" /          ; \    reverse solidus U+005C
-              "b" /          ; b    backspace       U+0008
-              "f" /          ; f    form feed       U+000C
-              "n" /          ; n    line feed       U+000A
-              "r" /          ; r    carriage return U+000D
-              "t" /          ; t    tab             U+0009
-              "u" 4HEXDIG )  ; uXXXX                U+XXXX
-
+index-selector      = "[" (quoted-member-name / element-index) "]"
+~~~~
 
-dq-unescaped = %x20-21 / %x23-5B / %x5D-10FFFF
```

Comment:
In the next PR, I'd prefer if we could move all the expression language parts over to an expression language section.